src/Chainweb/Pact/PactService.hs

```diff
@@ -648,51 +672,51 @@ rewindTo mb act = do
 
     failNonGenesisOnEmptyDb = fail "impossible: playing non-genesis block to empty DB"
 
-    playFork cp bhdb payloadDb newH parentHash (lastBlockHeight, lastHash) =
-      flip catches exHandlers $ do
-          parentHeader <- liftIO $ lookupM bhdb parentHash
```

Comment:
If `lookupM bdhdb parentHash` throw an `TreeDbKeyNotFound` exception, the handler below would call call `playFork` again with the same value for `parentHash`, which would result in an infinite recursion.

The new code only recurses on lookup failures for `lastHash`, in `findValidParent`.